src/unix/aix.c

```diff
@@ -886,16 +886,20 @@ int uv_set_process_title(const char* title) {
 
 int uv_get_process_title(char* buffer, size_t size) {
   size_t len;
-  len = strlen(process_argv[0]);
   if (buffer == NULL || size == 0)
     return UV_EINVAL;
-  else if (size <= len)
-    return UV_ENOBUFS;
 
   uv_once(&process_title_mutex_once, init_process_title_mutex_once);
   uv_mutex_lock(&process_title_mutex);
 
-  memcpy(buffer, process_argv[0], len + 1);
+  len = strlen(process_argv[0]);
+  if (size <= len) {
+    uv_mutex_unlock(&process_title_mutex);
+    return UV_ENOBUFS;
+  }
+
+  memcpy(buffer, process_argv[0], len);
```

Comment:
Good candidate for `strscpy` from https://github.com/libuv/libuv/pull/2065

Reply:
thanks @refack - do you mean to propose to wait till that lands and make that amendments, or propose it as a future enhancement when 2065 lands? please let me know.

Reply:
I'm not blocking in case this is of higher priority or #2065 takes time.
If #2065 land promptly, then personally I'd rather we use it in this PR.

Reply:
thanks, waiting has the advantage of avoiding one extra PR and reducing churn. not waiting has the advantage that it will make AIX green in CIs, after some time.

I will leave it up to other reviewers to discuss and decide.

Reply:
I'm good with landing this as is, FWIW. `uv__strscpy()` is for copying strings you don't know the size of ahead of time, but we know that here.